File: src/main/java/id/putra/wms/shared/base/entity/Customer.java
package id.putra.wms.shared.base.entity;

import java.io.Serializable;

import id.putra.wms.module.customer.model.entity.ContactPersonCustomer;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Entity
@Table(name = "customers")
@Data
@EqualsAndHashCode(callSuper = false)
public class Customer extends BaseEntity implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String code;
    private String name;
    private Boolean isActive;

    @ManyToOne
    @JoinColumn(name = "contact_person_id")
    private ContactPersonCustomer contactPerson;
}

File: web/types/customer.ts
export type ContactPersonCustomer = {
  id: number | null;
  name: string;
  email: string;
  phone: string;
  address: string;
};

export type Customer = {
  id: number | null;
  name: string;
  code: string;
  is_active: boolean;
  contactPerson?: ContactPersonCustomer;
};
